packages/task/src/browser/task-service.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from "inversify";
+import { ILogger } from '@theia/core/lib/common';
+import { FrontendApplication } from '@theia/core/lib/browser';
+import { ITaskServer, ITaskOutputEntryFoundEvent, ITaskExitedEvent } from '../common/task-protocol';
+import { TerminalWidget, TerminalWidgetOptions } from '@theia/terminal/lib/browser/terminal-widget';
+import { WidgetManager } from '@theia/core/lib/browser/widget-manager';
+import { TaskWatcher } from '../common/task-watcher';
+import { MessageService } from '@theia/core/lib/common/message-service';
+import { ProblemMarker } from '@theia/markers/lib/common/problem-marker';
+import { WorkspaceService } from '@theia/workspace/lib/browser/workspace-service';
+
+export const TaskTerminalWidgetFactory = Symbol('TaskTerminalWidgetFactory');
+export interface TaskTerminalWidgetFactory {
+    (options: TerminalWidgetOptions): TerminalWidget;
+}
+
+@injectable()
+export class TaskService {
+
+    constructor(
+        @inject(FrontendApplication) protected readonly app: FrontendApplication,
+        @inject(ITaskServer) protected readonly taskServer: ITaskServer,
+        @inject(ILogger) protected readonly logger: ILogger,
+        @inject(WidgetManager) protected readonly widgetManager: WidgetManager,
+        @inject(TaskTerminalWidgetFactory) protected readonly terminalWidgetFactory: TaskTerminalWidgetFactory,
+        @inject(TaskWatcher) protected readonly taskWatcher: TaskWatcher,
+        @inject(MessageService) protected readonly messageService: MessageService,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService
+    ) {
+        this.taskWatcher.onTaskOutputEntryFound((event: ITaskOutputEntryFoundEvent) => {
+            if (ProblemMarker.is(event.entry)) {
```

Comment:
I have removed this, since we do not yet generate markers in the backend. Will be back as part of the patch that will add parsing/handling of tasks output.